server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,117 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Function} eatFood Function to check that snake's next movement will be eating food.
+   * @param  {Array} rivalBody Rival snake's body coordinates
```

Comment:
Yes, A snake will have a rival body. But in the start it will be initial, I am still thinking about it. But for now, I am going with this design approach.

Reply:
rival body means? Can snake have two bodies, one own and another rival? And if that's the case then there are two snakes

Reply:
Yes, In a game there are two snakes on the board. By the way, now I am handling this thing by passing the game state object in the constructor.